plugins/hydra_ray_launcher/tests/test_ray_aws_launcher.py

```diff
@@ -0,0 +1,312 @@
+# Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved
+import hashlib
+import logging
+import os
+import random
+import string
+import subprocess
+import sys
+import tempfile
+from pathlib import Path
+from typing import Generator, List, Optional
+
+import pkg_resources
+import pytest
+from hydra.core.plugins import Plugins
+from hydra.plugins.launcher import Launcher
+from hydra.test_utils.launcher_common_tests import (
+    IntegrationTestSuite,
+    LauncherTestSuite,
+)
+from hydra.test_utils.test_utils import chdir_hydra_root, chdir_plugin_root
+from omegaconf import OmegaConf
+
+from hydra_plugins.hydra_ray_launcher._launcher_util import (  # type: ignore
+    _run_command,
+    ray_down,
+    ray_new_dir,
+    ray_rsync_up,
+    ray_up,
+)
+from hydra_plugins.hydra_ray_launcher.ray_aws_launcher import (  # type: ignore
+    RayAWSLauncher,
+)
+
+temp_remote_dir = "/tmp/hydra_test/"  # nosec
+temp_remote_wheel_dir = "/tmp/wheels/"  # nosec
+sweep_dir = "tmp_pytest_dir"  # nosec
+
+cluster_name = "IntegrationTest-" + "".join(
+    [random.choice(string.ascii_letters + string.digits) for n in range(5)]
+)
+
+win_msg = "Ray doesn't support Windows."
+cur_py_version = (
+    f"{sys.version_info.major}.{sys.version_info.minor}.{sys.version_info.micro}"
+)
+ami = os.environ.get("AWS_RAY_AMI", "")
+security_group_id = os.environ.get("AWS_RAY_SECURITY_GROUP", "")
+subnet_id = os.environ.get("AWS_RAY_SUBNET", "")
+instance_role = os.environ.get("INSTANCE_ROLE_ARN", "")
+
+# temporary solution. we will build customized AMI everytime the requirments changes.
+ray_requirements_sha = "bdefc02ff3b9f667f37ecce9f2fc5ac6"
+hydra_core_requirements_sha = "394984911320e5884cb055a5d13d7545"
+
+assert (
+    ami != "" and security_group_id != "" and subnet_id != "" and instance_role != ""
+), (
+    f"Missing variable for test, ami={ami}, security_group_id={security_group_id}, subnet_id={subnet_id}, "
+    f"instance_role={instance_role}"
+)
+
+
+ray_nodes_conf = {
+    "InstanceType": "m5.large",
+    "ImageId": ami,
+    "SubnetId": f"{subnet_id}",
+    "SecurityGroupIds": [security_group_id],
+    "IamInstanceProfile": {"Arn": instance_role},
+}
+
+ray_nodes_conf_override = str(ray_nodes_conf).replace("'", "").replace(" ", "")
+
+log = logging.getLogger(__name__)
+
+chdir_plugin_root()
+
+
+def build_ray_launcher_wheel(tmpdir: str) -> List[str]:
+    """
+    This  only works on ray launcher plugin wheels for now, reasons being in our base AMI
+    we do not necessarily have the dependency for other plugins.
+    """
```

Comment:
1. Are we expecting users to use our AMI or build their own?
2. Is that expectation documented?